### PR TITLE
[dotnet] Add support for workloads

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Add a `Microsoft.NET.Workload.<platform>` directory which contains
  `WorkloadManifest.json` and `WorkloadManifest.targets` required for the
  workload resolver.
   * For testing, this directory is installed into our local dotnet directory
     (`dotnet/sdk-manifests/5.0.100/Microsoft.NET.Workload.<platform>`).
* Convert `Microsoft.<platform>.Sdk` to an SDK workload pack.
* Replace `Sdk.props` with `AutoImport.props`, and include the default
  inclusion itemgroups definitions from here.
* Split default inclusion logic in two:
   * The itemgroups that import the default items has been moved into a props
     file and included from `AutoImport.props`. They're also conditioned
     based on the TargetPlatformIdentifier.
   * The rest of the logic (to enable default inclusion) is left in the
     .targets file.
* Merge several .props files into .targets files to follow the convention that
  .props files should be loaded before the main csproj, and .targets files
  after (these .props files were loaded after, so to avoid confusion the
  simplest solution was to merge their contents into the applicable .targets
  file and delete the .props files).
* Bundle the workload (WorkloadManifest.json/WorklooadManifest.targets) and
  the sdk + ref packs into an installable macOS pkg (and upload these
  packages).
* Updates tests to import `Microsoft.NET.Sdk` instead of
  `Microsoft.<platform>.Sdk` (this also required a Touch.Unit bump).
* Install the sdk + ref packs into the local dotnet directory (using symlinks
  to our on-disk nuget representation). This is required for local testing.
* Add some documentation.
* Finally fix a couple of bugs that these changes uncovered.

This PR might be easier to review commit-by-commit.

Ref: https://github.com/xamarin/xamarin-android/pull/5195

CC @pjcollins

### CODE REVIEW

File: dotnet/Makefile [hunk 1]
```diff
@@ -170,9 +173,55 @@ WORKLOAD_TARGETS += \
 	$(DOTNET_PACKS_PATH)/Microsoft.$1.Sdk/$2 \
 	$(DOTNET_PACKS_PATH)/Microsoft.$1.Ref/$2
 endef
+$(foreach platform,$(DOTNET_PLATFORMS),$(eval $(call InstallWorkload,$(platform),$($(platform)_NUGET_VERSION_NO_METADATA))))
+
+# We create three packages: one for the workload, one for the sdk package, and one for the ref package,
+# and then bundle them into a single pkg for distribution (named Microsoft.<platform>.Bundle.<version>.pkg).
+# This way we can create and publish separate updated for each workload/sdk/ref package later if we want to.
```
Comment (spouliot) [inline dotnet/Makefile:184]: `updated` -> `updates` ?

**Review by spouliot — APPROVED** [review]
anything required to notarize the new pkg ?

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-9897/1/display/redirect) (on internal Jenkins) succeeded

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-9897/1//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-9897/1//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-14.7.0.118.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/xamarin.ios-14.7.0.118.pkg)
* [Microsoft.iOS.Ref.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Ref.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh9897.1241+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh9897.1241+sha.3bfc112bb.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9897.1241+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9897.1241+sha.3bfc112bb.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9897.1241+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9897.1241+sha.3bfc112bb.nupkg)
* [Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9897.617+sha.3bfc112bb.nupkg)
* [Microsoft.iOS.Bundle.14.0.100-ci.pr.gh9897.617.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.iOS.Bundle.14.0.100-ci.pr.gh9897.617.pkg)
* [Microsoft.macOS.Bundle.10.15.100-ci.pr.gh9897.1241.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.macOS.Bundle.10.15.100-ci.pr.gh9897.1241.pkg)
* [Microsoft.tvOS.Bundle.14.0.100-ci.pr.gh9897.617.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.tvOS.Bundle.14.0.100-ci.pr.gh9897.617.pkg)
* [Microsoft.watchOS.Bundle.7.0.100-ci.pr.gh9897.617.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/Microsoft.watchOS.Bundle.7.0.100-ci.pr.gh9897.617.pkg)
* [delete-xma-build-host.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/package/delete-xma-build-host.pkg)

</details>

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/jenkins-results/generator-diff/index.html) (no change)
✅ [Test run succeeded](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/3bfc112bb5bbc78966b05a569ba912a20cf633c1/1/jenkins-results/tests/index.html)

File: dotnet/PackageInfo.plist [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<!--
+
+	This file is contains a list of app bundles contained with the package
```
Comment (tj-devel709) [inline dotnet/PackageInfo.plist:6]:
```suggestion
	This file contains a list of app bundles contained with the package
```

**Review by tj-devel709 — APPROVED** [review]

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-9897/2/display/redirect) (on internal Jenkins) succeeded

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-9897/2//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-9897/2//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-14.7.0.119.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/xamarin.ios-14.7.0.119.pkg)
* [Microsoft.iOS.Ref.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Ref.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Runtime.ios-arm.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Runtime.ios-arm64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Runtime.ios-x64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Runtime.ios-x86.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Sdk.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh9897.1242+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh9897.1242+sha.033681552.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9897.1242+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9897.1242+sha.033681552.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9897.1242+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9897.1242+sha.033681552.nupkg)
* [Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.tvOS.Ref.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.tvOS.Runtime.tvos-arm64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.tvOS.Runtime.tvos-x64.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.tvOS.Sdk.14.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.watchOS.Ref.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.watchOS.Runtime.watchos-arm.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.watchOS.Runtime.watchos-x86.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.watchOS.Sdk.7.0.100-ci.pr.gh9897.618+sha.033681552.nupkg)
* [Microsoft.iOS.Bundle.14.0.100-ci.pr.gh9897.618.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.iOS.Bundle.14.0.100-ci.pr.gh9897.618.pkg)
* [Microsoft.macOS.Bundle.10.15.100-ci.pr.gh9897.1242.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.macOS.Bundle.10.15.100-ci.pr.gh9897.1242.pkg)
* [Microsoft.tvOS.Bundle.14.0.100-ci.pr.gh9897.618.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.tvOS.Bundle.14.0.100-ci.pr.gh9897.618.pkg)
* [Microsoft.watchOS.Bundle.7.0.100-ci.pr.gh9897.618.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/Microsoft.watchOS.Bundle.7.0.100-ci.pr.gh9897.618.pkg)
* [delete-xma-build-host.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/package/delete-xma-build-host.pkg)

</details>

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/jenkins-results/generator-diff/index.html) (no change)
✅ [Test run succeeded](http://xamarin-storage/jenkins/xamarin-macios/PR-9897/03368155283cc1289e33d900bba317785dd51334/2/jenkins-results/tests/index.html)

**Pull request merged**